app/Jobs/WikiEntityImportJob.php

```diff
@@ -0,0 +1,240 @@
+            );
+        }
+    }
+
+    private static function acquireCredentials(string $wikiDomain): OAuthCredentials
+    {
+        $response = Http::withHeaders(['host' => $wikiDomain])->asForm()->post(
+            getenv('PLATFORM_MW_BACKEND_HOST').'/w/api.php?action=wbstackPlatformOauthGet&format=json',
+            [
+                'consumerName' => 'WikiEntityImportJob',
+                'ownerOnly' => '1',
+                'consumerVersion' => '1',
+                'grants' => 'basic|highvolume|import|editpage|editprotected|createeditmovepage|uploadfile|uploadeditmovefile|rollback|delete|mergehistory',
+                'callbackUrlTail' => '/w/index.php',
+            ],
+        );
+
+        if ($response->status() > 399) {
+            throw new \Exception('Unexpected status code '.$response->status().' from Mediawiki');
+        }
+
+        $body = $response->json();
+        if (!$body || $body['wbstackPlatformOauthGet']['success'] !== '1') {
+            throw new \ErrorException('Unexpected error acquiring oauth credentials for wiki '.$wikiDomain);
+        }
+
+        return OAuthCredentials::unmarshalMediaWikiResponse($body);
+    }
+}
+
+class TransferBotKubernetesJob
+{
+    public function __construct(
+        public Client $kubernetesClient,
+        public Wiki $wiki,
+        public OAuthCredentials $creds,
+        public array $entityIds,
+        public string $sourceWikiUrl,
+        public string $targetWikiUrl,
+        public int $importId,
+    ){
+        $this->kubernetesNamespace = Config::get('wbstack.api_job_namespace');
+    }
+
+    private string $kubernetesNamespace;
+
+    public function spawn(): string
+    {
+        $spec = $this->constructSpec();
+        $jobSpec = new KubernetesJob($spec);
+
+        $this->kubernetesClient->setNamespace($this->kubernetesNamespace);
+        $jobObject = $this->kubernetesClient->jobs()->apply($jobSpec);
+        $jobName = data_get($jobObject, 'metadata.name');
+        if (data_get($jobObject, 'status') === 'Failure' || !$jobName) {
+            // The k8s client does not fail reliably on 4xx responses, so checking the name
+            // currently serves as poor man's error handling.
+            throw new \RuntimeException(
+                'transferbot creation for wiki "'.$this->wiki->domain.'" failed with message: '.data_get($jobObject, 'message', 'n/a')
+            );
+        }
+        return $jobName;
+    }
+
+    private function constructSpec(): array
+    {
+        return [
+            'metadata' => [
+                'generateName' => 'run-transferbot-',
+                'namespace' => $this->kubernetesNamespace,
+                'labels' => [
+                    'app.kubernetes.io/instance' => $this->wiki->domain,
+                    'app.kubernetes.io/name' => 'run-transferbot',
+                ]
+            ],
+            'spec' => [
+                'ttlSecondsAfterFinished' => 0,
+                'backoffLimit' => 0,
```

Comment:
This means we only ever retry once and leave doing this up to the user that triggered a failed job.